backend/index.js

```diff
@@ -44,71 +56,133 @@ io.on("connection", (socket) => {
     socket.to(data.room).emit("receive_message", data);
   });
 
-	socket.on("send_image", (data) => {
-		console.log("Image received from", data.author, data.room);
+  socket.on("send_image", (data) => {
+    console.log("Image received from", data.author, data.room);
     socket.to(data.room).emit("receive_image", data);
     console.log("Image sent to", data.room);
-	});
+  });
 
   socket.on("disconnect", () => {
     console.log("A user has left the platform");
   });
 });
 
+// Custom Auth Middleware
+const authorization = (req, res, next) => {
+  const token = req.cookies.access_token;
+  if (!token) {
+    return res
+      .status(403)
+      .json({ message: "Proper authentication not supplied" });
+  }
+  try {
+    const data = jwt.verify(token, JWT_SECRET);
+    req.room = data.room;
+    req.room = data.role;
+    return next();
+  } catch {
+    return res.status(403);
+  }
+}
+
 // REST Endpoints
 app.get('/', (req, res) => {
-  res.json({
-    rooms: roomStore
-  })
+  return res
+    .status(200)
+    .json({
+      rooms: roomStore
+    });
 });
 
 app.get('/login', async (req, res) => {
   const username = req.query.username;
   const password = req.query.password;
+
+  const token = jwt.sign({ room: username, role: 'admin' }, JWT_SECRET);
   await client.get(username).then(storedPassword => {
     if (storedPassword == null) {
-      res.json({
-        authenticated: false,
-        reason: "No user found"
-      });
+      res
+        .status(403)
+        .json({
+          authenticated: false,
+          reason: "No user found"
+        });
       return;
     }
     if (storedPassword == password) {
-      res.json({
-        authenticated: true
-      });
-      return;
+      res
+        .cookie("access_token", token, {
+          secure: process.env.NODE_ENV === 'production',
+          sameSite: 'none'
```

Comment:
Look into samsite config

Reply:
We should leave this for a later PR implementing security fixes